packages/hardhat-chai-matchers/src/reverted/revertedWithCustomError.ts

```diff
@@ -160,7 +162,29 @@ export async function revertedWithCustomErrorWithArgs(
 
   for (const [i, actualArg] of Object.entries(actualArgs) as any) {
     const expectedArg = expectedArgs[i];
-    if (Array.isArray(expectedArg)) {
+    if (typeof expectedArg === "function") {
+      const errorPrefix = `The user-defined predicate for custom error argument #${
```

Comment:
The error prefix in `emit` is a bit different (it does not say "user-defined")

Reply:
Thanks. My first draft had "user-defined", but later I scrapped that, but I neglected to follow up in the reverts. Good catch. Addressed in https://github.com/NomicFoundation/hardhat/pull/2695/commits/c46da127ee74e3931bb5ae84c44c62f01e0aba72